File: src/entry.cpp
//:__________________________________________________________________
//  oQ3  |  Copyright (C) Ivan Mar (sOkam!)  |  GNU GPLv3 or later  :
//:__________________________________________________________________
#include "./id3.hpp"

//______________________________________
// @section Engine: Entry Point
//______________________________________
i32 main (i32 const argc, cstr const argv[argc]) {
  id3::Game game;
  id3::Engine engine = id3::Engine(game, argc, argv);
  engine.start();
  return id3::Error::ShouldNeverHappen;  // @note The engine has an internal termination process using jmp.
} //:: main

File: butcher/zig/entry.zig
//:__________________________________________________________________
//  oQ3  |  Copyright (C) Ivan Mar (sOkam!)  |  GNU GPLv3 or later  :
//:__________________________________________________________________
//! @fileoverview                                                 |
//!  Engine Debug Example                                         |
//!  Will run the engine with the default QuakeIIIArena Gamecode  |
//________________________________________________________________|
// @deps std
const std = @import("std");
// @deps id3
const id3 = @import("./id3.zig");


//______________________________________
// @section Entry Point
//____________________________
pub fn main () !u8 {
  var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
  defer arena.deinit();
  const A = arena.allocator();

  const game = id3.Game.create(.{});
  var engine = try id3.Engine.init(game, A);
  engine.start();
  return error.ShouldNeverHappen;  // @note The engine has an internal termination process.
}

